taxonomy-editor-frontend/src/pages/project/search/MultipleSelectFilter.tsx

```diff
@@ -0,0 +1,78 @@
+import {
+  FormControl,
+  OutlinedInput,
+  Checkbox,
+  Select,
+  MenuItem,
+  InputLabel,
+  ListItemText,
+} from "@mui/material";
+import { ChangeEvent, Dispatch, SetStateAction } from "react";
+
+type MultipleSelectFilterType = {
+  label: string;
+  filterValue: string[];
+  listOfChoices: string[];
+  mapCodeToValue: (code: string) => string;
+  mapValueToCode: (value: string) => string;
+  setQ: Dispatch<SetStateAction<string>>;
+  keySearchTerm: string;
+  setCurrentPage: Dispatch<SetStateAction<number>>;
+};
+
+export const MultipleSelectFilter = ({
+  label,
+  filterValue,
+  listOfChoices,
+  mapCodeToValue = () => "",
+  mapValueToCode = () => "",
+  setQ,
+  keySearchTerm,
+  setCurrentPage,
+}: MultipleSelectFilterType) => {
+  const handleChange = (
+    event: ChangeEvent<HTMLInputElement>,
+    languageCodeItem: string
+  ) => {
+    setCurrentPage(1);
+    if (!filterValue.includes(languageCodeItem)) {
+      setQ((prevQ) => prevQ + ` ${keySearchTerm}:${languageCodeItem}`);
+    } else {
+      setQ((prevQ) =>
+        prevQ.replace(`${keySearchTerm}:${languageCodeItem}`, "")
+      );
+    }
+    event.target.closest("Menu")?.dispatchEvent(new Event("close"));
```

Comment:
Is there maybe a more elegant way to handle it (using a controlled state maybe)?